debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java

```diff
@@ -79,9 +80,7 @@ protected SnapshotContext prepare(ChangeEventSourceContext context) throws Excep
 
     @Override
     protected Set<TableId> getAllTableIds(RelationalSnapshotContext ctx) throws Exception {
-        return jdbcConnection.getAllTableIds(ctx.catalogName);
-        // this very slow approach(commented out), it took 30 minutes on an instance with 600 tables
-        // return jdbcConnection.readTableNames(ctx.catalogName, null, null, new String[] {"TABLE"} );
+        return jdbcConnection.readTableNames(ctx.catalogName, null, null, null);
```

Comment:
So why is it that removed comment isn't a concern any more?

Reply:
Rather than introduce this special Oracle-specific `getAllTableIds` method, I overrode `readTableNames` with the same logic in the `OracleConnection` class.